### PR TITLE
add initial content to Documentation index

### PR DESCRIPTION
**Pull request by elmiko**
this describes the basic expectations for documentation when adding a
new component.

### CODE REVIEW

**Comment by elmiko** [comment]
@jeana-redhat we talked about this a little during team standup, curious if you might have any thoughts or comments about the instruction i am proposing.

**Review by JoelSpeed — COMMENTED** [review]
Is it worth mentioning enhancements and the role they play? I would have thought a good enhancement would also form the basis of the product docs, though maybe that comes from a different page within this project 🤔

**Comment by elmiko** [comment]
we do have a separate doc on creating the enhancement, https://github.com/elmiko/ocp-infrastructure-provider-onboarding-guide/blob/main/docs/openshift-enhancement/index.md
do you think it's worth making a mentioning and dropping a link in this doc?

File: docs/documentation/index.md [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
-TODO
+All OpenShift components that are released from Red Hat will have accompanying
+documentation in the [OpenShift product documentation][ocp-docs]. The content
+for these product documents are created and maintained by Red Hat, but they
+are inspired by the repository level documentation associated with the component.
+
+## Repository level documentation
+
+It is expected that any component which has its code in a Git repository will have
+documentation in that repository to cover its usage and troubleshooting. This
```
Comment (jeana-redhat) [inline docs/documentation/index.md:11]: If you do want to make this more of a hard requirement, perhaps:
```suggestion
Any component that has its code in a Git repository must have
documentation in that repository to cover its usage and troubleshooting. This
```

**Review by jeana-redhat — COMMENTED** [review]
Per Slack discussion :slightly_smiling_face:

File: docs/documentation/index.md (hunk 1 above)
Comment (elmiko) [reply docs/documentation/index.md:11]: i like that, thanks

**Comment by elmiko** [comment]
thanks for the suggestions, i updated the content to reference the enhancement (with link to doc page), and added the stronger language about repo docs.

File: docs/documentation/index.md [hunk 2]
```diff
@@ -1,3 +1,29 @@
 # Documentation
 
-TODO
+All OpenShift components that are released from Red Hat will have accompanying
+documentation in the [OpenShift product documentation][ocp-docs]. The content
+for these product documents are created and maintained by Red Hat, but they
+are inspired by the [OpenShift enhancement](../openshift-enhancement),
+repository level documentation, and direct discussions with the component
```
Comment (lobziik) [inline docs/documentation/index.md:7]: Hm, I'm thinking about some examples of good repo level documentation which we like, would nice be refer something as example.

Comment (elmiko) [reply docs/documentation/index.md:7]: i think that's a good idea, was there an example doc that you were thinking about?

we could point to the MAO documentation, but that's quite verbose. perhaps the [alibaba readme](https://github.com/openshift/cluster-api-provider-alibaba/blob/main/README.md), that seems to have some nice info about how to deploy and use the controller. wdyt?

Comment (elmiko) [reply docs/documentation/index.md:7]: maybe [cloud provider ibm](https://github.com/openshift/cloud-provider-ibm/blob/master/README.md) too?

Comment (lobziik) [reply docs/documentation/index.md:7]: Alibaba looks nice, lets reference it there?

Our MAPI docs is also good enough example i think

Comment (elmiko) [reply docs/documentation/index.md:7]: sounds good, i'll add some link. thanks!

**Comment by elmiko** [comment]
updated to add some example links

**Comment by elmiko** [comment]
this has gone over a week with no further review. i am going to merge under lazy consensus

**Pull request merged**